mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/definition/annotation/impl/AnnotationPojoTypeNodeMetadataContributorImpl.java

```diff
@@ -0,0 +1,278 @@
+	}
+
+	private <A extends Annotation> void addBridge(PojoNodeMappingCollector collector, A annotation) {
+		AnnotationBridgeBuilder<? extends Bridge, A> builder = createBridgeBuilder( annotation );
+		builder.initialize( annotation );
+		collector.bridge( builder );
+	}
+
+	private void addField(PojoPropertyNodeMappingCollector collector, PropertyModel<?> propertyModel, Field annotation) {
+		String cleanedUpFieldName = annotation.name();
+		if ( cleanedUpFieldName.isEmpty() ) {
+			cleanedUpFieldName = null;
+		}
+
+		BridgeBuilder<? extends FunctionBridge<?, ?>> builder = createFunctionBridgeBuilder( annotation, propertyModel );
+
+		collector.functionBridge( builder, cleanedUpFieldName, new AnnotationFieldModelContributor( annotation ) );
+	}
+
+	private void addIndexedEmbedded(PojoPropertyNodeMappingCollector collector, PropertyModel<?> propertyModel,
+			IndexedEmbedded annotation) {
+		String cleanedUpPrefix = annotation.prefix();
+		if ( cleanedUpPrefix.isEmpty() ) {
+			cleanedUpPrefix = null;
+		}
+
+		Integer cleanedUpMaxDepth = annotation.maxDepth();
+		if ( cleanedUpMaxDepth.equals( -1 ) ) {
+			cleanedUpMaxDepth = null;
+		}
+
+		String[] includePathsArray = annotation.includePaths();
+		Set<String> cleanedUpIncludePaths;
+		if ( includePathsArray.length > 0 ) {
+			cleanedUpIncludePaths = new HashSet<>();
+			Collections.addAll( cleanedUpIncludePaths, includePathsArray );
+		}
+		else {
+			cleanedUpIncludePaths = null;
+		}
+
+		collector.indexedEmbedded( cleanedUpPrefix, annotation.storage(), cleanedUpMaxDepth, cleanedUpIncludePaths );
+	}
+
+	private <A extends Annotation> AnnotationMarkerBuilder<A> createMarkerBuilder(A annotation) {
+		MarkerMapping markerMapping = annotation.annotationType().getAnnotation( MarkerMapping.class );
+		MarkerMappingBuilderReference builderReferenceAnnotation = markerMapping.builder();
+		BeanReference builderReference =
+				toBeanReference(
+						builderReferenceAnnotation.name(),
+						builderReferenceAnnotation.type(),
+						MarkerMappingBuilderReference.UndefinedImplementationType.class
+				)
+						.orElseThrow( () -> log.missingBuilderReferenceInBridgeMapping( annotation.annotationType() ) );
+
+		// TODO check generic parameters of builder.getClass() somehow, maybe in a similar way to what we do in FunctionBridgeUtil
+		return beanResolver.resolve( builderReference, AnnotationMarkerBuilder.class );
+	}
+
+	private BridgeBuilder<? extends IdentifierBridge<?>> createIdentifierBridgeBuilder(
+			DocumentId annotation, PropertyModel<?> propertyModel) {
+		IdentifierBridgeBeanReference bridgeReferenceAnnotation = annotation.identifierBridge();
+		Optional<BeanReference> bridgeReference = toBeanReference(
+				bridgeReferenceAnnotation.name(),
+				bridgeReferenceAnnotation.type(),
+				IdentifierBridgeBeanReference.UndefinedImplementationType.class
+		);
+		IdentifierBridgeBuilderBeanReference builderReferenceAnnotation = annotation.identifierBridgeBuilder();
```

Comment:
Maybe rename the variables `bridgeBuilder...`?

Reply:
I thought the name was already long enough, and "bridge" would be redundant because the whole method is about building an identifier bridge builder. I mean, if we go down this path I could also name it `identifierBridgeBuilder...`...
But as you wish.